cli/internal/providers/datacatalog/state/trackingplan.go

```diff
@@ -479,6 +480,55 @@ func (args *TrackingPlanPropertyArgs) PropertyByLocalID(id string) *TrackingPlan
 	return nil
 }
 
+// getAdditionalPropertiesDefaultVal returns the default value for additional properties based on the property type
+// the default value for additional properties is true for the following cases -
+// 1. Object type properties
+// 2. Multi-type properties with one of them "object" and no "array" type included
+// 3. Array properties with no item types defined
+// 4. Array properties where item type is object
+// 5. Array properties with multiple item types where at least one is object
+func getAdditionalPropertiesDefaultVal(prop *localcatalog.TPEventProperty) bool {
```

Comment:
Please add testcase for this function ?

Reply:
this is being tested indirectly via tests in state/trackingplan_test.go..added a few additional testcases in the PR as well(ex - testcase that checks if an array of objects is processed correctly)